invenio_cli/commands/steps.py

```diff
@@ -7,24 +7,37 @@
 
 """Invenio module to ease the creation and management of applications."""
 
-from ..helpers.process import run_interactive
-
+from ..helpers.process import run_interactive, ProcessResponse
 
 class FunctionStep(object):
     """A step which execution is a function call.
 
     Is composed of a function, arguments, and a message (feedback).
     """
 
-    def __init__(self, func, args=None, message=None):
+    def __init__(self, func, args=None, message=None, skippable=False):
         """Constructor."""
         self.func = func
         self.args = args or {}
         self.message = message
+        self.skippable = skippable
 
     def execute(self):
         """Execute the function with the given arguments."""
-        return self.func(**self.args)
+        try:
+            f = self.func(**self.args)
+            if self.skippable and f.status_code == 1:
+                f.status_code = 0
+                print(f"Warning: step failed {f.error}")
+                return f
+            else:
+                return f
+        except Exception as e:
```

Comment:
@fenekku I'm not super sure about this. Basically I build the ProcessResponse there (otherwise we'd have a void response from the call) with a successful status code but with `error` set to the warning message.

Reply:
@fenekku Thanks for detailed explanation. I think that's definitely a more solid approach. I'm working on it.